### ISSUE TITLE
Graphical User Interfaces

### ISSUE THREAD

**Issue by BobMuenchen** [matloff/R-vs.-Python-for-Data-Science#10]
An area where R has a clear edge is in the availability of point-and-click Graphical User Interfaces. I prefer to code, but I'm generally part of a research team where at least half the members prefer to use menus & dialog boxes. I like that I can put my R code behind their dialogs, or take the code their GUI writes and modify it. For my comparison of the GUI options for R, see http://r4stats.com/articles/software-reviews/r-gui-comparison/. I'm unaware of any GUIs for Python yet, but if you know of any, please let me know.

**Comment by matloff** [comment]
There are tons of IDEs for Python. Personally, I don't use IDEs in any language, but yes, in teams of mixed coding levels, they are needed.

**Comment by smartgamer** [comment]
Without IDE how do you see the figures instantly.  I want to learn.



On Jun 14, 2019, at 1:26 PM, Norm Matloff <notifications@github.com<mailto:notifications@github.com>> wrote:


There are tons of IDEs for Python. Personally, I don't use IDEs in any language, but yes, in teams of mixed coding levels, they are needed.

**Comment by BobMuenchen** [comment]
Yes, tons of IDEs that let you type code, but no GUIs (that I know of) that let you choose an analysis like a regression from a menu, drag variables to assign their roles, choose options, and click "OK" to complete the task without the user knowing code. The GUI then writes the R code and executes it without the user needing to see it. The code is then optionally saved in various ways (e.g. text or template) for reproducibility.

**Comment by matloff** [comment]
Re smartgamer's question: Start up R in a terminal window, NOT from an IDE, and type 'hist(Nile)'.

**Comment by matloff** [comment]
Yep, Python is not a Data Science language, so there is no Data Science GUI that I know of. I'm not sure that such a thing is a good idea, though, even in R.
